### PR TITLE
[VEN-1731]: VIP for HAY and SD Reward Distributors

### PR DESCRIPTION
**Pull request by narayanprusty**
## Description

Resolves VEN

### CODE REVIEW

File: vips/vip-162/vip-162-testnet.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          ["1240079365079365"]
+        ],
+      },
+      {
+        target: SD_REWARDS_DISTRIBUTOR,
+        signature: "setRewardTokenSpeeds(address[],uint256[],uint256[])",
+        params: [
+          [MARKET_BNBx],
+          ["1157407407407407"],
+          ["1157407407407407"]
```
Comment (GitGuru7) [inline vips/vip-162/vip-162-testnet.ts:90]:
```suggestion
          ["1240079365079365"],
          ["1240079365079365"]
```

File: vips/vip-162/vip-162.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          ["1240079365079365"]
+        ],
+      },
+      {
+        target: SD_REWARDS_DISTRIBUTOR,
+        signature: "setRewardTokenSpeeds(address[],uint256[],uint256[])",
+        params: [
+          [MARKET_BNBx],
+          ["1157407407407407"],
+          ["1157407407407407"]
```
Comment (GitGuru7) [inline vips/vip-162/vip-162.ts:90]:
```suggestion
          ["1240079365079365"],
          ["1240079365079365"]
```

File: vips/vip-162/vip-162-testnet.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+const HAY_AMOUNT = "2000000000000000000000";
+const SD_AMOUNT = "2000000000000000000000";
+const MARKET_BNBx = "0x644A149853E5507AdF3e682218b8AC86cdD62951";
+const MARKET_HAY = "0x170d3b2da05cc2124334240fB34ad1359e34C562";
+
+export const vip162Testnet = () => {
+  const meta = {
+    version: "v2",
+    title: "VIP-162 HAY and SD Rewards",
+    description: `Uppgrade comptroller to support rewards distribution for duplicate reward tokens`,
```
Comment (Debugger022) [inline vips/vip-162/vip-162-testnet.ts:22]:
```suggestion
    description: `Upgrade comptroller to support rewards distribution for duplicate reward tokens`,
```

File: vips/vip-162/vip-162.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+const HAY_AMOUNT = "2000000000000000000000";
+const SD_AMOUNT = "2000000000000000000000";
+const MARKET_BNBx = "0x5E21bF67a6af41c74C1773E4b473ca5ce8fd3791";
+const MARKET_HAY = "0xCa2D81AA7C09A1a025De797600A7081146dceEd9";
+
+export const vip162 = () => {
+  const meta = {
+    version: "v2",
+    title: "VIP-162 HAY and SD Rewards",
+    description: `Uppgrade comptroller to support rewards distribution for duplicate reward tokens`,
```
Comment (Debugger022) [inline vips/vip-162/vip-162.ts:22]:
```suggestion
    description: `Upgrade comptroller to support rewards distribution for duplicate reward tokens`,
```

**Review by Debugger022 — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/163

**Review by chechu — APPROVED** [review]

**Pull request merged**